packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1419,6 +1422,24 @@ abstract class BuildableElement extends Element {
     markNeedsBuild();
   }
 
+  bool _hadDependenciesWhenDeactivated = false;
```

Comment:
Can we use the _dirty bool to represent this information?  Basically, have the deactivated elements who need a build when reactivated marked as dirty?